libs/barista-components/secondary-nav/src/section/secondary-nav-section.ts

```diff
@@ -131,6 +112,42 @@ export class DtSecondaryNavSection {
 
   constructor(private readonly _changeDetectorRef: ChangeDetectorRef) {}
 
+  ngAfterContentInit(): void {
+    this._checkForActiveStates();
+  }
+
+  ngOnDestroy(): void {
+    this._destroy$.next();
+    this._destroy$.complete();
+  }
+
+  /** @internal Check if sections has any active links or groups with active links within them */
+  _checkForActiveStates(): void {
+    // Check if section has any active links, and activate + expand the entire section
+    // Includes a setTimeout((), 0) because the RouterLinkActive directive uses Promise.resolve().then(())
+    // https://github.com/angular/angular/blob/master/packages/router/src/directives/router_link_active.ts#L125
+    setTimeout(() => {
```

Comment:
Nice, thanks for the extra comment here. This will make it easier to understand in the future, why this was necessary.